### PR TITLE
Cache implementation for for SAP use case.

### PR DESCRIPTION
**Pull request by sa-progress**
Obvious fix; these changes are the result of automation not creative thinking.

Signed-off-by: sa-progress <samir.anand@progress.com>

### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec AWS](https://github.com/inspec/inspec-aws/CONTRIBUTING.md) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [ ] All Unit Tests pass
- [ ] `rake lint` passes
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_security_groups.rb [hunk 1]
```diff
@@ -25,6 +25,9 @@ class AwsSecurityGroups < AwsResourceBase
              .register_column(:group_names, field: :group_name)
              .register_column(:vpc_ids,     field: :vpc_id)
              .register_column(:tags,        field: :tags)
+             .register_column(:descriptions, field: :description)
```
Comment (sathish-progress) [inline libraries/aws_security_groups.rb:28]: we can use the new plural resource methods here

File: libraries/aws_security_group.rb [hunk 2]
```diff
@@ -12,48 +12,53 @@ class AwsSecurityGroup < AwsResourceBase
...
 
   def initialize(opts = {})
     opts = { group_id: opts } if opts.is_a?(String)
     opts[:group_id] = opts.delete(:id) if opts.key?(:id) # id is an alias for group_id
 
     super(opts)
-    validate_parameters(require_any_of: %i(group_id group_name vpc_id))
 
-    filter = []
-    if opts.key?(:vpc_id)
```
Comment (sathish-progress) [inline libraries/aws_security_group.rb:25]: why are these removed?

File: libraries/aws_backend.rb [hunk 3]
```diff
@@ -454,7 +454,7 @@ def is_permissions_error(error)
   end
 
   def map_tags(tag_list)
-    return {} if tag_list.nil? || tag_list.empty?
+    return {} if tag_list.nil? || tag_list.empty? || tag_list.is_a?(Hash)
```
Comment (sathish-progress) [inline libraries/aws_backend.rb:457]: why are we returning empty `{}` when tag_list is already an Hash?

File: libraries/aws_security_group.rb (hunk 2 above)
Comment (sa-progress) [reply libraries/aws_security_group.rb:25]: Its not removed .Since these lines were moved below It is showing removed.

**Review by sathish-progress — APPROVED** [review]
LGTM

**Review by IanMadd — APPROVED** [review]
